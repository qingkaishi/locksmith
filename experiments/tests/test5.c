#include <pthread.h>
#include <stdio.h>

struct {
	int data;
} foo;
int shared = 1;
pthread_mutex_t t;

void * dostuff(void * a) {
	pthread_mutex_lock(&t);
	foo.data++;
	shared++;
	pthread_mutex_unlock(&t);
}
int main() {
  int i = 1;
	pthread_t t1;
	pthread_mutex_init(&t,NULL);

	for(i = 0; i<10; i++) {
		pthread_create(&t1, NULL, dostuff, NULL);
	}
}

