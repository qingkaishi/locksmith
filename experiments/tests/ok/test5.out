************************* STARTING *************************

shared: &shared:test5.c:7
        alloc.data:test5.c:4
&shared:test5.c:7 is protected by:
  concrete t:test5.c:19
  t:test5.c:8
alloc.data:test5.c:4 is protected by:
  concrete t:test5.c:19
  t:test5.c:8
*************************** DONE ***************************
